presto-main/src/main/java/com/facebook/presto/operator/PagesIndexStats.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.Block;
+import io.airlift.stats.CounterStat;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class PagesIndexStats
+{
+    private final CounterStat totalBlocks = new CounterStat();
```

Comment:
`CounterStat` is quite expensive. I was working on an improvement (https://github.com/highker/airlift/tree/decay) but can't convince myself the improvement is a win. How about just use atomic data types? @martint

Reply:
@highker : Yep. A atomic long is (mostly) enough in this case.